sdk/core/azure-core/azure/core/pipeline/transport/pyodide.py

```diff
@@ -0,0 +1,160 @@
+from . import HttpRequest
+from ._requests_asyncio import AsyncioRequestsTransport
+
+
+class PyodideTransportResponse(AsyncHttpResponseImpl):
+    """Async response object for the `PyodideTransport`."""
+
+    def __init__(self, **kwargs):
+        super(PyodideTransportResponse, self).__init__(**kwargs)
+        # clone to avoid reading from the same `FetchResponse` a second time in `load_body`.
+        self._js_stream = self.internal_response.clone().js_response.body
+        self._js_reader = None
+
+    async def close(self) -> None:
+        """We don't actually have control over closing connections in the browser, so we just pretend
+        to close.
+        """
+        self._is_closed = True
+
+    async def load_body(self) -> None:
+        """Load the body of the response."""
+        if self._content is None:
+            self._content = await self._internal_response.bytes()
+
+    def body(self) -> bytes:
+        """The body is just the content."""
+        return self.content
+
+class PyodideStreamDownloadGenerator(AsyncIterator):
+    """Simple stream download generator that returns the contents of
+    a request.
+    """
+
+    # pylint: disable=unused-argument
+    def __init__(self, pipeline: Pipeline, response: PyodideTransportResponse, *_, **kwargs):
+        self._block_size = response._block_size
+        self.response = response
+        # use this to efficiently store bytes.
+        if kwargs.pop("decompress", False):
+            self._js_reader = response._js_stream.pipeThrough(js.DecompressionStream.new("gzip")).getReader()
+        else:
+            self._js_reader = response._js_stream.getReader()
+        self._stream = BytesIO()
+
+        self._closed = False
+        # We cannot control how many bytes we get from `response.reader`. `self.buffer_left`
+        # indicates how many unread bytes there are in `self.stream`
+        self.buffer_left = 0
+        self.done = False
+
+    async def __anext__(self) -> bytes:
+        """Get the next block of bytes."""
+        if self._closed:
+            raise StopAsyncIteration()
+
+        # remember the initial stream position
+        start_pos = self._stream.tell()
+        # move stream position to the end
+        self._stream.read()
+        # read from reader until there is no more data or we have `self._block_size` unread bytes.
+        while self.buffer_left < self._block_size:
+            read = await self._js_reader.read()
+            if read.done:
+                self._closed = True
+                break
+            self.buffer_left += self._stream.write(bytes(read.value))
+
+        # move the stream position back to where we started
+        self._stream.seek(start_pos)
+        self.buffer_left -= self._block_size
+        return self._stream.read(self._block_size)
+
+class PyodideTransport(AsyncioRequestsTransport):
+    """Implements a basic HTTP sender using the Pyodide Javascript Fetch API.
```

Comment:
you can mark this transport as provisional in docstrings, [this](https://github.com/Azure/azure-sdk-for-python/tree/azure-core_1.19.1/sdk/core/azure-core/azure/core/rest) is what I did for documentation when `azure.core.rest` was still provisional (just search provisional)